### PR TITLE
Added fuse_rms_norm lowering

### PR DESCRIPTION
**Pull request by cehongwang**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project (You can use the linters)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/dynamo/lowering/passes/replace_fused_rms_norm.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            if user.op == "call_function" and user.target == operator.getitem:
+                # If the getitem is extracting the first element (the output tensor)
+                if not x_normalized.meta:
+                    x_normalized.meta = copy.copy(node.meta)
+                user.replace_all_uses_with(x_normalized)
+                gm.graph.erase_node(user)
+
+    gm.graph.erase_node(node)
+
+    return x_normalized
```
Comment (apbose) [inline py/torch_tensorrt/dynamo/lowering/passes/replace_fused_rms_norm.py]: In the replacement with the normalized, in line 80, should we also check for the second return value of the op? The rstd as we see in the test cases. Though yeah I think application wise, its mainly used for the gradient, but to be consistent with the op signature

File: py/torch_tensorrt/dynamo/lowering/passes/replace_fused_rms_norm.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            )
+
+        x_normalized.meta = {}
+
+        for user in list(node.users):
+            if user.op == "call_function" and user.target == operator.getitem:
+                # If the getitem is extracting the first element (the output tensor)
+                if not x_normalized.meta:
+                    x_normalized.meta = copy.copy(node.meta)
+                user.replace_all_uses_with(x_normalized)
```
Comment (zewenli98) [inline py/torch_tensorrt/dynamo/lowering/passes/replace_fused_rms_norm.py]: As [Naren previously mentioned](https://github.com/pytorch/TensorRT/pull/4001#discussion_r2696630581), can you add a log here when each node is changed?

File: py/torch_tensorrt/dynamo/lowering/passes/replace_fused_rms_norm.py (hunk 1 above)
Comment (zewenli98) [reply py/torch_tensorrt/dynamo/lowering/passes/replace_fused_rms_norm.py]: yeah I think it should have two Tensor outputs per https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/native_functions.yaml

File: py/torch_tensorrt/dynamo/lowering/passes/_aten_lowering_pass.py [hunk 3]
```diff
@@ -3,8 +3,11 @@
 from typing import Any, Callable, Optional, Sequence, Union
 
 import torch
+from torch._inductor.fx_utils import get_node_storage
```
Comment (narendasan) [inline py/torch_tensorrt/dynamo/lowering/passes/_aten_lowering_pass.py]: Remove this line, its unused now.

**Comment by cehongwang** [comment]
Pending: update the FakeTensorUpdater once this PR goes in:
https://github.com/pytorch/pytorch/pull/159523

**Comment by narendasan** [comment]
@lanluo-nvidia @cehongwang if we are waiting on this from PyTorch, then can we lift the fused_rms_norm support from 2.10 into 2.11?

**Comment by cehongwang** [comment]
I think we can. There is already a previous version of FakeTensorUpdater merged in 2.10
